developer/developer-guide.md

```diff
@@ -1,3 +1,872 @@
+the use of the jar artifact instead of the amp to do the new install. 
+The JAR has the format `apix-impl-{ALFRESCO-VERSION}-{APIX-VERSION}.jar` and can be found under 
+`apix-impl/{ALFRESCO-VERSION}/build/libs/`, where `ALFRESCO-VERSION` is one of *(50|51|52|60|61|62)*.
+The new installation can be done either through the DE web interface, or with the following gradle task.
+```bash
+./gradlew :apix-impl:apix-impl-{ALFRESCO-VERSION}:installBundle -Phost={ALFRESCO-HOST} -Pport={ALFRESCO-PORT}
+```
+
+*Protip:* If you get tired of changing the port after every `docker-compose up`, you can temporarily put a
+fixed port in the *docker-compose.yml* of the version you are working with. (The rationale behind using 
+variable ephemeral ports is that during parallel builds on Jenkins port clashes must be avoided.)
+
+For example for version 5.1, change in *apix-docker/51/docker-compose.yml* 
+the ports line from:
+```yaml
+services:
+  alfresco-core:
+    ports:
+      - ${DOCKER_IP}:8080
+``` 
+to: 
+```yaml
+services:
+  alfresco-core:
+    ports:
+      - ${DOCKER_IP}:9051:8080
+```
+and then restart the containers with:
+
+```bash
+./gradlew :apix-docker:docker-51:composeUp --info
+```
+
+# Alfred api Concepts
+Alfred API is composed of two logical layers.
+
+* The base layer is a *Java API* built on top of the Alfresco.
+* A *REST API* is built on top of the Java abstraction layer, exposing a stable HTTP API.
+
+Instead of mirroring the Alfresco API one-to-one, Alfred API groups frequently used operations
+together in a single method call. This allows you to focus on business concerns instead of focusing
+on fetching all required data from Alfresco.
+
+For example, the `NodeService.getMetadata()` method returns an object with all metadata about 
+a node: its type, aspects and properties in a single function call. It groups together the 
+information that would otherwise have to be obtained by combining requests for type, aspects 
+and properties separately.
+
+## Data objects
+Alfred API has data objects that mirror the Alfresco concepts of QName, NodeRef, StoreRef, Path, 
+ContentData and ContentInputStream. These data objects are used to communicate with the 
+Alfred Java API without being dependent on Alfresco data types.
+
+Conversion between Alfresco and Alfred API data objects is the responsibility of the 
+`ApixToAlfrescoConversion` service. It is also possible to construct an Alfred API data object by
+passing its string representation to the constructor.
+
+## REST API
+The Alfred REST API is a thin wrapper around the Java abstraction layer. It converts its received
+parameters to the corresponding Alfred API data objects, then calls the corresponding service and
+serializes its return value to JSON.
+
+### Developer notes on search syntax
+#### Unimplemented search options:
+These options have some code towards handling, but are not implemented such that they are used in the search.
+* `facets.mincount`
```

Comment:
This list does not render in the PDF nor in the website. I think an extra leading newline will help.